tests/e2e/scale/test_scale_pvc_pod_multi_project.py

```diff
@@ -0,0 +1,141 @@
+        cephfs_sc_obj = helpers.default_storage_class(constants.CEPHFILESYSTEM)
+        pvc_count_each_itr = 5
+        size = f"{random.randrange(5, 105, 5)}Gi"
+        fio_size_param = '2G'
+        fio_rate_param = '16k'
+        self.namespace_list, all_pod_obj = ([] for i in range(2))
+
+        # Pre-requisite check for number of OSD's and app worker nodes.
+        helpers.add_required_osd_count(total_osd_nos=3)
+
+        # Create machineset for app worker nodes, which will create one app worker node
+        self.ms_name = machine.create_custom_machineset(instance_type='m5.4xlarge', zone='a')
+        machine.wait_for_new_node_to_be_ready(self.ms_name)
+        self.app_worker_nodes = machine.get_machine_from_machineset(self.ms_name)
+
+        # Create namespace
+        self.namespace_list.append(helpers.create_project())
+
+        # Continue to iterate till the scale pvc limit is reached
+        while True:
+            if scale_count <= len(all_pod_obj):
+                log.info(f"Scaled {scale_count} pvc and pods")
+                break
+            else:
+                log.info(f"Create {pvc_count_each_itr} pods & pvc")
+                pod_obj, pvc_obj = helpers.create_multi_pvc_pod(
+                    self.namespace_list[-1].name, rbd_sc_obj, cephfs_sc_obj, pvc_count_each_itr,
+                    size, fio_rate=fio_rate_param, start_io=start_io, fio_size=fio_size_param,
+                    fio_runtime=3600, node_selector=constants.SCALE_NODE_SELECTOR
+                )
+                all_pod_obj.extend(pod_obj)
+                try:
+                    # Check enough resources available in the dedicated app workers
+                    if helpers.add_worker_based_on_cpu_utilization(
+                        machineset_name=self.ms_name, node_count=1, expected_percent=75,
+                        role_type='app,worker'
+                    ):
+                        logging.info(f"Nodes added for app pod creation")
+                    else:
+                        logging.info(f"Existing resource are enough to create more pods")
+
+                    # Check for ceph cluster OSD utilization
+                    if not cluster.validate_osd_utilization(osd_used=80):
+                        logging.info("Cluster OSD utilization is below 80")
+                    else:
+                        raise CephHealthException("Cluster OSDs are near full")
+
+                    # Change fio_size and fio_rate param according to env
+                    fio_size_param = helpers.suggest_io_size_based_on_cls_usage()
+                    fio_rate_param = helpers.suggest_fio_rate_based_on_cls_iops()
+
+                    # Check for pg_balancer
+                    if cluster.validate_pg_balancer():
+                        logging.info("OSD consumption and PG distribution is good to continue")
+                    else:
+                        raise UnexpectedBehaviour("Unequal PG distribution to OSDs")
```

Comment:
it would be good to move this condition at the end. When new osd's are added backfilling starts and it takes some time and the test fails in between.

Reply:
ahh correct, need to skip this if there is rebalance going on in cluster.